airflow/sensors/base.py

```diff
@@ -281,6 +282,16 @@ def run_duration() -> float:
         self.log.info("Success criteria met. Exiting.")
         return xcom_value
 
+    def resume_execution(self, next_method: str, next_kwargs: dict[str, Any] | None, context: Context):
+        try:
+            return super().resume_execution(next_method, next_kwargs, context)
+        except (AirflowException, TaskDeferralError) as e:
+            if self.soft_fail:
+                raise AirflowSkipException(e)
```

Comment:
Do we need to cast the (wrapped) exception to a str first?

Reply:
Actually it works in both ways, but it's a bit confusing. I will replace by str(e)